mdata/cwr.go

```diff
@@ -1,23 +1,78 @@
 package mdata
 
 import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
 	"time"
 
-	"github.com/grafana/metrictank/mdata/chunk"
 	"github.com/raintank/schema"
 )
 
 // ChunkWriteRequest is a request to write a chunk into a store
+//go:generate msgp
 type ChunkWriteRequest struct {
-	Metric    *AggMetric
+	Callback  func() `msg:"-"`
 	Key       schema.AMKey
-	Chunk     *chunk.Chunk
 	TTL       uint32
-	Span      uint32
+	T0        uint32
+	Data      []byte
 	Timestamp time.Time
 }
 
 // NewChunkWriteRequest creates a new ChunkWriteRequest
-func NewChunkWriteRequest(metric *AggMetric, key schema.AMKey, chunk *chunk.Chunk, ttl, span uint32, ts time.Time) ChunkWriteRequest {
-	return ChunkWriteRequest{metric, key, chunk, ttl, span, ts}
+func NewChunkWriteRequest(callback func(), key schema.AMKey, ttl, t0 uint32, data []byte, ts time.Time) ChunkWriteRequest {
+	return ChunkWriteRequest{callback, key, ttl, t0, data, ts}
+}
+
+// MdWithCwrs is used by the whisper importer utilities to transfer the data
+// from the reader to the writer
+//go:generate msgp
```

Comment:
We dont have to use msgp for this.   The majority of the data is already in []byte slices.  so there is not a large overhead in marshalling/unmarshalling.  We could just use `gob` if we wanted.

Reply:
using msgp seems simple though. what would be the advantage of using gob? that we don't need to generate the msgp methods for the structs in github.com/raintank/schema ?

Reply:
basically.  Given msgp is already working, there is probably no point in replacing it.